reconfiguration/include/reconfiguration/pruning_handler.hpp

```diff
@@ -17,22 +17,111 @@
+  // well as providing read-only information to the operator node.
+  //
+  // The following configuration options are honored by the state machine:
+  //  * pruning_enabled - a system-wide configuration option that indicates if
+  //  pruning is enabled for the replcia. If set to false,
+  //  LatestPrunableBlockRequest will return 0 as a latest block(indicating no
+  //  blocks can be pruned) and PruneRequest will return an error. If not
+  //  specified, a value of false is assumed.
+  //
+  //  * pruning_num_blocks_to_keep - a system-wide configuration option that
+  //  specifies a minimum number of blocks to always keep in storage when pruning.
+  //  If not specified, a value of 0 is assumed. If
+  //  pruning_duration_to_keep_minutes is specified too, the more conservative
+  //  pruning range of the two options will be used (the one that prunes less
+  //  blocks).
+  //
+  //  * pruning_duration_to_keep_minutes - a system-wide configuration option that
+  //  specifies a time range (in minutes) from now to the past that determines
+  //  which blocks to keep and which are older than (now -
+  //  pruning_duration_to_keep_minutes) and can, therefore, be pruned. If not
+  //  specified, a value of 0 is assumed. If pruning_num_blocks_to_keep is
+  //  specified too, the more conservative pruning range of the two will be used
+  //  (the one that prunes less blocks). This option requires the time service to
+  //  be enabled.
+  //
+  // The LatestPrunableBlockRequest command returns the latest block ID from the
+  // replica's storage that is safe to prune. If no blocks can be pruned, 0 is
+  // returned.
+  //
+  // The PruneRequest command prunes blocks from storage by:
+  //  - verifying the PruneRequest message's signature
+  //  - verifying that the number of LatestPrunableBlock messages in the
+  //  PruneRequest is equal to the number of replicas in the system
+  //  - verifying the signatures of individual LatestPrunableBlock messages in the
+  //  PruneRequest.
+  // If all above conditions are met, the state machine will prune blocks from the
+  // genesis block up to the the minimum of all the block IDs in
+  // LatestPrunableBlock messages in the PruneRequest message.
  public:
-  bool handle(const concord::messages::LatestPrunableBlockRequest&, concord::messages::LatestPrunableBlock&) override {
-    return true;
-  }
-  bool handle(const concord::messages::PruneStatusRequest&, concord::messages::PruneStatus&) override { return true; }
-  bool handle(const concord::messages::PruneRequest&, kvbc::BlockId&) override { return true; }
+  // Construct by providing an interface to the storage engine, state transfer,
+  // configuration and tracing facilities. Note this constructor may throw an
+  // exception if there is an issue with the configuration (for example, if the
+  // configuration enables pruning but does not provide a purning operator
+  // public key).
+  PruningHandler(const kvbc::IReader &,
```

Comment:
Please convert all methods to start with a lowercase letter as per code conventions